### DISCUSSION TITLE
About the next release of alphaTab

### DISCUSSION THREAD

**Discussion by mxAlexG** [CoderLine/alphaTab#950] [Q&A] [answered]
Hi,
What are the plans for the next release of alphaTab and is it early to say when? 

I performed a build for the current develop branch and noticed some issues such as the cursor sometimes suddenly stopping and the playback audio not working properly (noise). I assume they're caused by other things within the project that are actively under development?

Also, are there plans for further improving MusicXML Importer? especially for the ornaments and articulations parts of it? It is great as it is but alphaTab is close to surpassing other libraries out there that are specific to MusicXML, so it would be counter intuitive to miss such an opportunity given how ubiquitous this format is becoming.

Thanks and regards.

**Comment by Danielku15** [comment, marked as answer]
> What are the plans for the next release of alphaTab and is it early to say when?

Generally it is hard to commit to any timelines for me. alphaTab is a pure hobby project of mine and any development I do in my spare time (during holidays, weekends and evenings). My private life has been very busy and I did not have much time this year to work on alphaTab. I have some contributors which started to make smaller features/bugfixes but beside that the project is fully maintained by me. 

The high level roadmap you can see here: https://github.com/orgs/CoderLine/projects?type=classic

I have the plan to make a 1.2.3 release where I bring various bugfixes and small improvements to the market. The 1.3. release was delayed so much because of some technical challenges targeting Android and iOS natively. While Kotlin claims to be great and support cross platform mobile workflows, the reality is rather disappointing. I had to wait often for months that they fixed issues and bugs in Kotlin and extend the platform to even support some things I needed. 

> Also, are there plans for further improving MusicXML Importer?

In this regards my plans are mainly driven by the "customer requirements". I have no direct plans currently to extend and advance the supported feature set unless people ask for it. MusicXML is still somehow in an experimental state because the format is quite a pain in many ways. Being an exchange format primarily, every application somehow generates stuff differently and expects the own layouting/engraving engine to actually display stuff correctly. 

If there are requests to support some dedicated missing features, I am happy to put them into the roadmap, otherwise I would rather focus on some other features people have been already asking for. The 1.4 release planned to have the theme around more and better file format support, hence there is a good chance that also the MusicXML importer gets some improvements and to reach more feature parity with the other formats.

My goal is usually: First support the full feature set of Guitar Pro. With those features typically new rendering and playback capabilities come into the core of alphaTab. Then I extend alphaTex to also allow writing these new music notation elements. And then the other importers (Capella, MusicXML, and the future planned ones) will get extended gradually to also fill the data model. 

The first part is typically something I am doing while the alphaTex and other importer extensions is something I also hope to activate contributors.
